### PR TITLE
Core: Improve Gasmodel charging, monomorphize to gas table.

### PR DESCRIPTION
**Pull request by jmcardon**
This PR monomorphizes our Gas model charging to always use our gas table. There's about a 10% speed increase from this change.

Numbers pre-pr:
```
benchmarking Pact Core Benchmarks/Pure Code/Factorial 1000
time                 216.1 μs   (215.8 μs .. 216.7 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 216.0 μs   (215.8 μs .. 216.8 μs)
std dev              1.236 μs   (327.0 ns .. 2.522 μs)

benchmarking Pact Core Benchmarks/Coin Transfer/InterpretOnly(transfer) from SenderA to SenderB using: CEK
time                 76.84 μs   (76.76 μs .. 76.92 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 76.95 μs   (76.91 μs .. 77.00 μs)
std dev              106.9 ns   (83.52 ns .. 159.7 ns)

benchmarking Pact Core Benchmarks/Coin Transfer/InterpretOnly(transfer) from SenderA to SenderB using: Direct
time                 75.01 μs   (74.97 μs .. 75.05 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 75.01 μs   (74.95 μs .. 75.05 μs)
std dev              108.6 ns   (83.82 ns .. 140.9 ns)
```

Post pr:
```
benchmarking Pact Core Benchmarks/Pure Code/Factorial 1000
time                 184.4 μs   (181.5 μs .. 188.9 μs)
                     0.998 R²   (0.995 R² .. 0.999 R²)
mean                 186.2 μs   (184.4 μs .. 188.3 μs)
std dev              6.571 μs   (5.282 μs .. 8.562 μs)
variance introduced by outliers: 32% (moderately inflated)

benchmarking Pact Core Benchmarks/Coin Transfer/InterpretOnly(transfer) from SenderA to SenderB using: CEK
time                 71.82 μs   (71.66 μs .. 72.01 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 71.89 μs   (71.76 μs .. 72.04 μs)
std dev              307.4 ns   (228.9 ns .. 414.1 ns)

benchmarking Pact Core Benchmarks/Coin Transfer/InterpretOnly(transfer) from SenderA to SenderB using: Direct
time                 69.09 μs   (68.96 μs .. 69.21 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 69.06 μs   (68.91 μs .. 69.23 μs)
std dev              329.2 ns   (214.6 ns .. 462.8 ns)
```

PR checklist:

* [x] Test coverage for the proposed changes
* [x] PR description contains example output from repl interaction or a snippet from unit test output
* [x] (If Relevant) Documentation has been (manually) updated at https://docs.kadena.io/pact

Additionally, please justify why you should or should not do the following:

* [x] Benchmark regressions
* [x] Confirm replay/back compat (Ignore until core release)
* [x] (For Kadena engineers) Run integration-tests against a Chainweb built with this version of Pact (Ignore until core release)

### CODE REVIEW

File: pact/Pact/Core/IR/Eval/Runtime/Utils.hs [hunk 1]
```diff
@@ -373,3 +374,117 @@ anyCapabilityBeingEvaluated
 anyCapabilityBeingEvaluated caps = do
   capsBeingEvaluated <- use (esCaps . csCapsBeingEvaluated)
   return $! any (`S.member` caps) capsBeingEvaluated
+
+readKeyset' :: i -> Text -> EvalM e b i (Maybe KeySet)
```
Comment (rsoeldner) [inline pact/Pact/Core/IR/Eval/Runtime/Utils.hs:378]: I would have not looked for it in this module, is there some reason for this reorg

File: pact/Pact/Core/SizeOf.hs [hunk 2]
```diff
@@ -23,7 +23,7 @@ module Pact.Core.SizeOf
   ( SizeOf(..)
   , SizeOf1(..)
   , constructorCost
-  , Bytes
+, Bytes
```
Comment (rsoeldner) [inline pact/Pact/Core/SizeOf.hs:26]: Accident

**Review by rsoeldner — APPROVED** [review]

File: pact/Pact/Core/Gas/Types.hs [hunk 3]
```diff
@@ -209,22 +198,19 @@ data CapOp
...
   -- ^ The cost of concatenating two elements
   -- TODO: We actually reuse this cost for construction as well for objects/lists. Should we
   -- instead consider renaming the objcat and listcat constructors to be ListCatOrConstruction
-  -- | GALinear !Word64 {-# UNPACK #-} !LinearGasArg
-  -- ^ Cost of linear-based gas
-  | GIntegerOpCost !IntegerPrimOp !Integer !Integer
-  -- ^ Cost of integer operations
   | GMakeList !Integer !Word64
   -- ^ Cost of creating a list of `n` elements + some memory overhead per elem
-  | GAApplyLam (Maybe FullyQualifiedName) !Int
```
[the remark this answers is not in the document]
Comment (jmcardon) [reply pact/Pact/Core/Gas/Types.hs:226]: We moved the constructor up. We didnt' stop charging for it.

File: pact/Pact/Core/IR/Eval/Runtime/Utils.hs (hunk 1 above)
Comment (jmcardon) [reply pact/Pact/Core/IR/Eval/Runtime/Utils.hs:378]: Good question

Initially I made some more aggressive changes to the gas model which actually impacted these functions,and I noticed the duplication across `Evaluator.hs` and `CoreBuiltin.hs`

**Pull request merged**